### PR TITLE
Add indication to admin UI of whether a report has been forwarded

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Reports list

In the lists of reports, an arrow (fontawesome's “share”) is displayed when the report is known to have been forwarded.

image

## Report

For reports about remote users, the table will show either of those according to the situation:
- Forwarded: Yes
- Forwarded: No
- Forwarded: Unknown (reported before the information was recorded)

image

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
What means forwarded?

**Comment by ThibG** [comment]
@SuperSandro2000 when a user reports an account that's on a remote server, they have the option to forward that report to the originating server as well. That's what it means.

**Comment by Gargron** [comment]
Maybe instead of "Yes", "Yes, %{domain} has been notified"? Other mods/admins absolutely will not know what "Forwarded" means, though I understand that has to be short to fit in the table column.

**Comment by ThibG** [comment]
Hm, I can do that, but I'm not sure what we should do for “No” then :/

**Comment by Gargron** [comment]
I'm not really sure what the utility of this information is. Like I guess "Maybe there is hope origin will take care of this before I have to"? But is that really so useful to know on the report overview page? Also, the forward icon looks a lot like the reply icon. People will not know what it means for sure.

**Comment by ThibG** [comment]
> I'm not really sure what the utility of this information is. Like I guess "Maybe there is hope origin will take care of this before I have to"?

I mean, I have seen this request a bunch of time. I guess to have an idea if the remote admins not doing anything is because they didn't want to, or because they weren't been made aware of it, for instance.

@nightpool was interested

> But is that really so useful to know on the report overview page? Also, the forward icon looks a lot like the reply icon. People will not know what it means for sure.

Well, it doesn't take up much space. Maybe another icon would be better suited but idk which. Thankfully there's a tooltip.

**Comment by Siphonay** [comment]
A situation arose on a server I’m moderating yesterday where we didn't feel like it was appropriate to take action ourselves and would rather know how the targeted instance's admins would deal with it. Knowing if a report was forwarded would had saved us asking about it to the admins or the users who made the report. I think such a function would be useful

**Comment by ProgVal** [comment]
Same as @Siphonay, so I know if I have to contact the remote admins.

Additionally, it would be nice to have an option for admins to forward the report even if the user initially didn't

**Comment by Psycojoker** [comment]
Hello,

As an admin/moderator having this information would help greatly to:

- know the intent of the member of our community: do they expect us to act locally or do they also expect this from the other instance
- know if the other instance if aware of that and so, if it's a serious case be inform on how they act on it
- this last information is important to check if this other instance has a moderation strategy similar to ours and if not, if we need to take further actions (like contacting them and/or escalate to more coercive actions)

As moderator of our communities, ensuring that our members are evolving as much as possible in a space that matches our CoC is very important, this kind of information will help us do a better job.

**Comment by StarryAster** [comment]
I think that could also be useful in situations where a forwarded report is unclear or in a different language, in a way that makes contacting the forwarded instance's admins about the report more urgent than if the report was just solved locally first.

This is pretty minor in comparison to other stated reasons, though.

**Comment by renatolond** [comment]
Passing by to say that I already do this manually (reaching out to users when the report is from another instance to see if they forwarded it or if, depending on the case, I should myself do a second report to forward to the original instance).
So this would help quite a lot :)

**Comment by ThibG** [comment]
@Gargron can we have that? There's clearly interest from multiple instance admins for that feature.

File: app/views/admin/reports/index.html.haml [hunk 1]
```diff
@@ -59,6 +59,10 @@
                 = fa_icon('camera')
                 = report.media_attachments.count
 
+              - if report.forwarded?
+                %span.report-card__summary__item__content__icon{ title: t('admin.reports.forwarded') }
+                  = fa_icon('share')
```
Comment (Gargron) [inline app/views/admin/reports/index.html.haml:64]: How about just text: "Forwarded to %{domain}"

File: config/locales/en.yml [hunk 2]
```diff
@@ -430,6 +430,8 @@ en:
       comment:
         none: None
       created_at: Reported
+      forwarded: Forwarded
+      forwarded_unknown: Unknown (reported before the information was recorded)
```
Comment (Gargron) [inline config/locales/en.yml:434]: Unnecessary detailedness. You can just write "not available" or reuse "n/a" from somewhere if we already have that. Just using `-` is also an option.

**Comment by ThibG** [comment]
Updated according to reviews:
image

File: app/views/admin/reports/show.html.haml [hunk 3]
```diff
@@ -46,6 +46,16 @@
         %td{ colspan: 2 }
           - if @report.action_taken?
             = table_link_to 'envelope-open', t('admin.reports.reopen'), admin_report_path(@report, outcome: 'reopen'), method: :put
+      - unless @report.target_account.local?
+        %tr
+          %th= t('admin.reports.forwarded')
```
Comment (Gargron) [inline app/views/admin/reports/show.html.haml:51]: I think this string is missing

Comment (ThibG) [reply app/views/admin/reports/show.html.haml:51]: Indeed :weary: 
Fixed.

**Review by Gargron — APPROVED** [review]

**Pull request merged**